crates/codegen/runtime/cargo/crate/src/runtime/parser/parser_support/parser_function.rs

```diff
@@ -57,16 +57,39 @@ where
             ParserResult::PrattOperatorMatch(..) => unreachable!("PrattOperatorMatch is internal"),
 
             ParserResult::NoMatch(no_match) => {
+                let mut trivia_nodes = if let ParserResult::Match(matched) =
+                    Lexer::leading_trivia(parser, &mut stream)
```

Comment:
Does this mean we are parsing the leading trivia again at this point?
I wonder if it is possible to just reuse the work already done by `no_match` here.

Reply:
The complexity lies in deciding what to do in choices `(This | That)`: which trivia's should be stored, `This` or `That`? In fact, trivia is in itself already parsed twice when considering `This` or `That`, so it doesn't make a big difference.

Reply:
And IIUC the trivia is cached already, so in terms of performance it shouldn't matter much

Reply:
I see. In that case, should we use `stream.mark()` and `stream.rewind()`, to make sure that we haven't advanced in the stream?

Reply:
Thanks for explaining. Can you please add a comment there explaining that? I think that is important context for the future.